packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -265,17 +287,42 @@ export class GarbageCollector implements IGarbageCollector {
         this.deleteTimeoutMs = this.gcOptions.deleteTimeoutMs ?? defaultDeleteTimeoutMs;
 
         let prevSummaryGCVersion: number | undefined;
+
         // GC can only be enabled during creation. After that, it can never be enabled again. So, for existing
         // documents, we get this information from the metadata blob.
         if (existing) {
             prevSummaryGCVersion = getGCVersion(metadata);
             // Existing documents which did not have metadata blob or had GC disabled have version as 0. For all
             // other exsiting documents, GC is enabled.
             this.gcEnabled = prevSummaryGCVersion > 0;
+
+            if (metadata?.sessionTimeoutMs !== undefined) {
+                // Grab from metadata
+                this.sessionExpiryMs = metadata.sessionTimeoutMs;
+                // TODO: Change to ClientSessionExpiredError
+                // this.sessionExpiryTimeout = setTimeout(() => this.closeFn(
+                //     new ClientSessionExpiredError(`Client expired, it has lasted ${this.sessionExpiryMs} ms.`,
+                //     this.sessionExpiryMs)),
+                //     this.sessionExpiryMs);
+            }
         } else {
             // For new documents, GC has to be exlicitly enabled via the gcAllowed flag in GC options.
             this.gcEnabled = gcOptions.gcAllowed === true;
+            this.gcSessionTimeoutEnabled = this.gcOptions.gcSessionTimeoutEnabled === true;
+            if (this.gcSessionTimeoutEnabled) {
+                // Only use this gcOption for testing.
+                const testExpiryMs = this.gcOptions.gcTestSessionTimeoutMs;
+                this.sessionExpiryMs = testExpiryMs ?? defaultContainerRuntimeExpiryMs;
+            }
+        }
+
+        if (this.sessionExpiryMs !== undefined) {
```

Comment:
Add a comment here saying that if session expiry is enabled, we need to close the container when the timeout expires (or something to that effect).